cylc/flow/task_pool.py

```diff
@@ -2109,48 +2119,91 @@ def _get_flow_nums(
-        """Assumes task is in the pool"""
-        # TODO is this flag still needed, and consistent with "cylc set"?
+    def _force_trigger(self, itask: 'TaskProxy', on_resume: bool = False):
+        """Process a manually triggered task, ready for job submission.
+
+        Assumes the task is in the pool.
+
+        Triggering a queued task will:
+          - run it, regardless of queue limiting
+
+        Triggering an unqueued task will:
+          - queue it, if the queue is limiting activity
+          - run it, if the queue is not limiting activity
+
+        After state reset and queue handling:
+        - if on_resume is False, add the task to tasks_to_trigger_now
+        - if on_resume is True, add the task to tasks_to_trigger_on_resume
+
+        The scheduler will release tasks from the tasks_to_trigger sets.
+
+        """
         itask.is_manual_submit = True
         itask.reset_try_timers()
+
         if itask.state_reset(TASK_STATUS_WAITING):
             # (could also be unhandled failed)
             self.data_store_mgr.delta_task_state(itask)
-        # (No need to set prerequisites satisfied here).
-        if itask.state.is_runahead:
-            # Release from runahead, and queue it.
-            self.rh_release_and_queue(itask)
+
+        if itask.state_reset(is_runahead=False):
+            # Can force trigger runahead-limited tasks.
+            self.data_store_mgr.delta_task_state(itask)
             self.spawn_to_rh_limit(
                 itask.tdef,
                 itask.tdef.next_point(itask.point),
                 itask.flow_nums
             )
-        else:
-            # De-queue it to run now.
-            self.task_queue_mgr.force_release_task(itask)
+
+        if not itask.state.is_queued:
+            # queue it if limiting
+            active, _ = self.count_active_tasks()
+            if self.task_queue_mgr.push_task_if_limited(itask, active):
+                itask.state_reset(is_queued=True)
+                self.data_store_mgr.delta_task_state(itask)
+
+        elif self.task_queue_mgr.remove_task(itask):
+            # else release it from the queue to run now
+            itask.state_reset(is_queued=False)
+            self.data_store_mgr.delta_task_state(itask)
+
+        if not itask.state.is_queued:
+            # If not queued now, record the task as ready to run.
+            itask.waiting_on_job_prep = True
+
+            if on_resume:
+                with suppress(KeyError):
+                    # In case previously triggered without --on-resume.
+                    # (It should have run already, but just in case).
+                    self.tasks_to_trigger_now.remove(itask)
+                self.tasks_to_trigger_on_resume.add(itask)
+            else:
+                with suppress(KeyError):
+                    # In case previously triggered with --on-resume.
+                    self.tasks_to_trigger_on_resume.remove(itask)
+                self.tasks_to_trigger_now.add(itask)
```

Comment:
```suggestion
                self.tasks_to_trigger_on_resume.add(itask)
                # In case previously triggered without --on-resume.
                # (It should have run already, but just in case).
                self.tasks_to_trigger_now.discard(itask)
            else:
                self.tasks_to_trigger_now.add(itask)
                # In case previously triggered with --on-resume.
                self.tasks_to_trigger_on_resume.discard(itask)
```

Reply:
Nice, thanks.